[CI] InferenceRestIT test {p0=inference/80_random_rerank_retriever/Random rerank retriever predictably shuffles results} failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #3812 / debian-11_platform-support-unix](https://gradle-enterprise.elastic.co/s/f4qrz7l5l7swm)
- [elasticsearch-pull-request #28862 / part-2](https://gradle-enterprise.elastic.co/s/s5l4lpwi4cxvs)
- [elasticsearch-pull-request #28802 / part-2](https://gradle-enterprise.elastic.co/s/gyxymk6vrbddi)
- [elasticsearch-periodic-platform-support #3797 / oraclelinux-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/fjcqtwx73uv3e)
- [elasticsearch-periodic-platform-support #3794 / amazonlinux-2023_platform-support-aws](https://gradle-enterprise.elastic.co/s/w2bu46kglogqw)
- [elasticsearch-periodic-platform-support #3791 / ubuntu-2004-aarch64_checkpart2_platform-support-arm](https://gradle-enterprise.elastic.co/s/uznnymh6wtolg)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:inference:yamlRestTest" --tests "org.elasticsearch.xpack.inference.InferenceRestIT.test {p0=inference/80_random_rerank_retriever/Random rerank retriever predictably shuffles results}" -Dtests.seed=1449F1BCAD7A1D83 -Dtests.locale=nl-NL -Dtests.timezone=Europe/Warsaw -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('test%20%7Bp0%3Dinference%2F80_random_rerank_retriever%2FRandom%20rerank%20retriever%20predictably%20shuffles%20results%7D'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.inference.InferenceRestIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: Failure at [inference/80_random_rerank_retriever:51]: 
Expected: a numeric value within <0.001> of <0.136>
     but: <0.2068796157836914> differed by <0.0698796157836914> more than delta <0.001>
```

**Issue Reasons:**
- [main] 6 failures in test test {p0=inference/80_random_rerank_retriever/Random rerank retriever predictably shuffles results} (1.6% fail rate in 378 executions)
- [main] 2 failures in step part-2 (2.1% fail rate in 96 executions)
- [main] 4 failures in pipeline elasticsearch-periodic-platform-support (50.0% fail rate in 8 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (2.1% fail rate in 96 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -80,9 +80,6 @@ tests:
 - class: org.elasticsearch.xpack.restart.CoreFullClusterRestartIT
   method: testSnapshotRestore {cluster=UPGRADED}
   issue: https://github.com/elastic/elasticsearch/issues/111799
-- class: org.elasticsearch.xpack.inference.InferenceRestIT
-  method: test {p0=inference/80_random_rerank_retriever/Random rerank retriever predictably shuffles results}
-  issue: https://github.com/elastic/elasticsearch/issues/111999
 - class: org.elasticsearch.smoketest.SmokeTestMultiNodeClientYamlTestSuiteIT
   issue: https://github.com/elastic/elasticsearch/issues/112147
 - class: org.elasticsearch.smoketest.WatcherYamlRestIT

--- a/x-pack/plugin/inference/src/yamlRestTest/resources/rest-api-spec/test/inference/80_random_rerank_retriever.yml
+++ b/x-pack/plugin/inference/src/yamlRestTest/resources/rest-api-spec/test/inference/80_random_rerank_retriever.yml
@@ -8,6 +8,8 @@ setup:
       indices.create:
         index: test-index
         body:
+          settings:
+            number_of_shards: 1
           mappings:
             properties:
               text:

